types/camelize/index.d.ts

```diff
@@ -7,14 +7,14 @@ type CamelCase<S extends string> = S extends `${infer FirstWord}${CamelSeparator
     ? `${Lowercase<FirstWord>}${Uppercase<SecondChar>}${CamelCase<Remaining>}`
     : Lowercase<S>;
 
-type Camelize<T> = {
-    [K in keyof T as CamelCase<string & K>]: T[K] extends {} ? Camelize<T[K]> : T[K];
-};
-
-declare function camelize<T>(obj: T): T extends string ? CamelCase<T>
+type Camelize<T> = T extends Date ? T
     : T extends RegExp ? T
-    : T extends Date ? T
-    : T extends {} ? Camelize<T>
+    : T extends Array<infer I> ? Array<Camelize<I>>
```

Comment:
I've also added support for arrays (these are [handled separately in the original package](https://github.com/ljharb/camelize/blob/main/index.js#L11) as well)